### PR TITLE
Py35 updates

### PR DESCRIPTION
**Pull request by newville**
This addresses #281, fixing use of function calls in constraint expressions for Py3.5
Also includes a couple small code cleanups.

### CODE REVIEW

File: lmfit/asteval.py [hunk 1]
```diff
@@ -622,25 +622,27 @@ def on_raise(self, node):    # ('type', 'inst', 'tback')
 
     def on_call(self, node):
         "function execution"
-        #  ('func', 'args', 'keywords', 'starargs', 'kwargs')
+        #  ('func', 'args', 'keywords', and 'starargs', 'kwargs' in py < 3.5)
```
Comment (Tillsten) [inline lmfit/asteval.py]: left ',' before and

Comment (Tillsten) [inline lmfit/asteval.py]: else it looks all good as far as i can tell.

**Comment by newville** [comment]
I'll merge this in about a day unless there are objections.

Also: I'm not sure why Travis CI didn't like Python 3.5 (no dateutil module?).   Tests do pass for Python3.5 for me.  Hopefully that can be figured out soon.

**Comment by tacaswell** [comment]
What does lmfit use dateutil for?  

It looks like the meta-data on pypi won't even let you install it via pip.  I guess you could do a source install...

**Comment by newville** [comment]
@tacaswell  Thanks -- good point.   dateutil is not used.  It's probably just cruft from some earlier travis recipe.  I'll try without it.

**Comment by newville** [comment]
@tacaswell  that worked ... thanks!

**Pull request merged**